### DISCUSSION TITLE
dss_python -  make the binaries available via the Python Package Index (PyPI)

### DISCUSSION THREAD

**Discussion by christopinka** [dss-extensions/dss-extensions#25] [Q&A] [answered]
Is this available yet? Could it be an alternative to having to do something like the script in the c# wrapper to get a tagged release version of c binaries for platform?

**Comment by PMeira** [comment, marked as answer]
> Is this available yet?

@christopinka The latest release? Yep, 16 hours ago: https://pypi.org/project/dss-python/#history

Just made the release announcement on GitHub though: https://github.com/dss-extensions/dss_python/releases/tag/0.14.2

If you meant in general, it's been on PyPI since 2018. The PyPI package name is `dss_python` (or `dss-python`, same thing) since a dead package already used the `dss` name, but from Python you `import dss` since that dead package is not related to OpenDSS at all.

By the way, we used to provide conda packages too but it's not worth the effort due to very low usage in comparison with PyPI/pip. We plan to submit to conda-forge later this year to automate that and better integrate with conda.

> Could it be an alternative to having to do something like the script in the c# wrapper to get a tagged release version of c binaries for platform?

Besides Linux ARM releases, all packages are already built on GitHub Actions for Python. For 0.14.2, the packages at the bottom of the pages https://github.com/dss-extensions/dss_python/actions/runs/5076128896 and https://github.com/dss-extensions/dss_python_backend/actions/runs/5075942460

We could automate the whole release process later, yes, but frankly sometimes we spend more time handling the CI issues than we would prefer. We'll probably look into that again after some major refactoring/porting is done on DSS C-API, since it will make things much easier.
